include/SDL3/SDL_main.h

```diff
@@ -562,30 +562,17 @@ extern SDL_DECLSPEC void SDLCALL SDL_GDKSuspendComplete(void);
 #include <SDL3/SDL_close_code.h>
 
 #if !defined(SDL_MAIN_HANDLED) && !defined(SDL_MAIN_NOIMPL)
-    /* include header-only SDL_main implementations */
-    #if defined(SDL_MAIN_USE_CALLBACKS) \
-        || defined(SDL_PLATFORM_WINDOWS) || defined(SDL_PLATFORM_IOS) || defined(SDL_PLATFORM_TVOS) \
-        || defined(SDL_PLATFORM_3DS) || defined(SDL_PLATFORM_NGAGE) || defined(SDL_PLATFORM_PS2) || defined(SDL_PLATFORM_PSP) \
-        || defined(SDL_PLATFORM_EMSCRIPTEN)
+    /* include header-only SDL_main implementations
+     * Note: currently Android is the only platform where we rename main() to SDL_main() but
+     *  do *not* use SDL_main_impl.h (because SDL_main() is called from external Java code).
+     *  If other platforms like that turn up, add them next to "defined(SDL_PLATFORM_ANDROID)"
+     */
+    #if ( defined(SDL_MAIN_USE_CALLBACKS) || defined(SDL_MAIN_NEEDED) || defined(SDL_MAIN_AVAILABLE) ) \
+        && ! ( defined(SDL_PLATFORM_ANDROID) )
```

Comment:
```suggestion
        && !defined(SDL_PLATFORM_ANDROID)
```

Reply:
this was intentional so adding another android-like platform is easier/less error prone (just add `|| defined(SDL_PLATFORM_BLA)`)

Reply:
Yeah, but it looks a little out of place and I trust whoever adds a platform can work through the && || logic necessary. :)

Reply:
ok, fixed this (and the other thing as well)